cmake/CompilerOptions.cmake

```
@@ -0,0 +1,176 @@
+#inspired from https://github.com/cginternals/glbinding/blob/master/cmake/CompileOptions.cmake
+
+#
+# Platform and architecture setup
+#
+
+# Get upper case system name
+string(TOUPPER ${CMAKE_SYSTEM_NAME} SYSTEM_NAME_UPPER)
+
+# Determine architecture (32/64 bit)
+set(X64 OFF)
+if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+    set(X64 ON)
+endif()
+
+
+#
+# Project options
+#
+
+set(RA_DEFAULT_PROJECT_OPTIONS
+    DEBUG_POSTFIX
+    "d"
+    POSITION_INDEPENDENT_CODE
+    ON
+    CXX_EXTENSIONS
+    Off
+    CXX_STANDARD
+    17
```

```suggestion
    17
    CXX_STANDARD_REQUIRED
    ON
```
This will prevent decay to older standard if the compiler does not support C++17.
see 
https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html#prop_tgt:CXX_STANDARD
and
https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD_REQUIRED.html#prop_tgt:CXX_STANDARD_REQUIRED